File: frontend/src/api/outbound.ts
import { apiRequest } from './client';

export interface OutboundInput { lot_id: number; location_id: number; qty: number; note: string }
export interface OutboundResult { lot_id: number; location_id: number; qty: number; movement_id: number }
export interface OutboundRecord extends OutboundInput {
  movement_id: number; lot_code: string; product_name: string; unit: string;
  location_code: string; actor_name: string; created_at: string;
}

export function submitOutbound(payload: OutboundInput): Promise<OutboundResult> {
  return apiRequest('/api/outbound', {
    method: 'POST', body: JSON.stringify(payload), signal: AbortSignal.timeout(15000),
  });
}

export function getOutboundRecords(): Promise<OutboundRecord[]> {
  return apiRequest('/api/outbound', { signal: AbortSignal.timeout(10000) });
}

File: backend/schemas/outbound.py
from typing import Annotated

from pydantic import BaseModel, ConfigDict, Field

PositiveInt = Annotated[int, Field(strict=True, gt=0, le=9223372036854775807)]


class OutboundCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    lot_id: PositiveInt
    location_id: PositiveInt
    qty: PositiveInt
    note: str = Field(default="", max_length=500)


class OutboundResult(BaseModel):
    lot_id: int
    location_id: int
    qty: int
    movement_id: int


class OutboundRecord(BaseModel):
    movement_id: int
    lot_id: int
    lot_code: str
    product_name: str
    unit: str
    location_id: int
    location_code: str
    qty: int
    actor_name: str
    note: str
    created_at: str
